### PR TITLE
Codestyle: Don't use auto where not warranted

### PR DESCRIPTION
**Pull request by akien-mga**
We allow using auto for lambdas or complex macros where a return type may change based on the parameters. But where the type is clear, we should be explicit.

https://www.youtube.com/watch?v=z87g8IQTVjw

### CODE REVIEW

File: tests/scene/test_arraymesh.h [hunk 1]
```diff
@@ -195,7 +195,7 @@ TEST_CASE("[SceneTree][ArrayMesh] Surface metadata tests.") {
 	}
 
 	SUBCASE("Returns correct format for the mesh") {
-		auto format = RS::ARRAY_FORMAT_BLEND_SHAPE_MASK | RS::ARRAY_FORMAT_TEX_UV | RS::ARRAY_FORMAT_INDEX;
+		RS::ArrayFormat format = RS::ARRAY_FORMAT_BLEND_SHAPE_MASK | RS::ARRAY_FORMAT_TEX_UV | RS::ARRAY_FORMAT_INDEX;
```
Comment (akien-mga) [inline tests/scene/test_arraymesh.h:198]: Example of why `auto` isn't good when the type _should_ be clear...
```
invalid conversion from 'int' to 'RenderingServer::ArrayFormat' [-fpermissive]
```

**Comment by AThousandShips** [comment]
What about:
https://github.com/godotengine/godot/blob/d6d8cb1a171b0aa74bc54e615e7c1e068af5a5d8/modules/openxr/scene/openxr_hand.cpp#L216-L217

**Comment by akien-mga** [comment]
I missed those, my regex was `\tauto `, I didn't think of `const`.

**Comment by AThousandShips** [comment]
Also:
https://github.com/godotengine/godot/blob/d6d8cb1a171b0aa74bc54e615e7c1e068af5a5d8/modules/openxr/openxr_api.cpp#L304

**Review by AThousandShips — APPROVED** [review]
LGTM 🎉

File: modules/openxr/openxr_api.cpp [hunk 2]
```diff
@@ -2077,7 +2076,7 @@ XRPose::TrackingConfidence _transform_from_location(const T &p_location, Transfo
 	Basis basis;
 	Vector3 origin;
 	XRPose::TrackingConfidence confidence = XRPose::XR_TRACKING_CONFIDENCE_NONE;
-	const auto &pose = p_location.pose;
+	const XrPosef &pose = p_location.pose;
```
Comment (akien-mga) [inline modules/openxr/openxr_api.cpp:2079]: I had to look up what types `T` can take, where it's defined in `openxr.h`, to know which type its `pose` member uses. This really should be explicit for the reader.

**Pull request merged**